src/Debug/skeletonViewer.ts

```diff
@@ -611,18 +612,18 @@ export class SkeletonViewer {
                     continue;
                 }
 
-                let boneAbsoluteRestTransform = new Matrix();
-                this.getAbsoluteRestPose(bone, boneAbsoluteRestTransform);
+                let boneAbsoluteBindPoseTransform = new Matrix();
+                this.getAbsoluteBindPoseToRef(bone, boneAbsoluteBindPoseTransform);
 
                 let anchorPoint = new Vector3();
 
-                boneAbsoluteRestTransform.decompose(undefined, undefined, anchorPoint);
+                boneAbsoluteBindPoseTransform.decompose(undefined, undefined, anchorPoint);
 
                 bone.children.forEach((bc, i) => {
-                    let childAbsoluteRestTransform : Matrix = new Matrix();
-                    bc.getRestPose().multiplyToRef(boneAbsoluteRestTransform, childAbsoluteRestTransform);
+                    let childAbsoluteRestPoseTransform : Matrix = new Matrix();
+                    bc.getRestPose().multiplyToRef(boneAbsoluteBindPoseTransform, childAbsoluteRestPoseTransform);

```

Comment:
Is it correct to use `getRestPose` here? In the helper function we use `getBindPose`. Is there a reason for not using the the same when figuring out the child bone position?

Reply:
@Pryme8  / @Popov72  / @CedricGuillemet

Reply:
Yes, as it is computing an absolute bind pose matrix it should start by using the local bind matrix, not the rest pose.

Reply:
Are you saying that it should be changed to `getBindPose`?

Reply:
yes